### ISSUE TITLE
WMTS: error downloading capabilities

### ISSUE THREAD

**Issue by sikmir** [tumic0/GPXSee#104]
### Steps to reproduce
1. Create WMTS mapsource file:
```xml
$ cat << EOF > ~/.gpxsee/maps/World/Europe/FI/MapAnt.xml
<?xml version="1.0" encoding="UTF-8"?>                                                                                                     
<map xmlns="http://www.gpxsee.org/map/1" type="WMTS">                                                                                      
  <name>MapAnt</name>                                                                                                                    
  <url>http://wmts.mapant.fi/wmts.php</url>                                                                                              
  <copyright>Map data: National Land Survey of Finland (CC BY 4.0) | Rendering: MapAnt (CC BY 4.0)</copyright>                           
  <zoom min="7" max="19"/>                                                                                                               
  <layer>MapAnt.fi</layer>                                                                                                  
  <set>ETRS-TM35FIN</set>                                                                                                                
</map>
EOF
```
### Actual behaviour
GPXSee failed to download capabilities:
```
(dbg) GET http://wmts.mapant.fi/wmts.php?service=WMTS&Version=1.0.0&request=GetCapabilities
(dbg) Reply error: 0
(dbg) GET wmts.xml
(dbg) Reply error: 301
Error downloading file: : Protocol "" is unknown
```
### RCA
```sh
$ curl -vL -o /dev/null 'http://wmts.mapant.fi/wmts.php?service=WMTS&Version=1.0.0&request=GetCapabilities'
> GET /wmts.php?service=WMTS&Version=1.0.0&request=GetCapabilities HTTP/1.1
> Host: wmts.mapant.fi
...
< HTTP/1.1 302 Found
< Location: wmts.xml
...
> GET /wmts.xml HTTP/1.1
> Host: wmts.mapant.fi
...
< HTTP/1.1 200 OK
```
So, something wrong with handling HTTP redirects from WMTS server.
### Configuration
GPXSee Version: latest master
Operating System: ArchLinux

**Comment by tumic0** [comment]
The server returns an invalid (according to RFC2616 it must be an absoluteURI) Location header: 

> HTTP/1.1 302 Found
> Date: Tue, 17 Apr 2018 22:27:59 GMT
> Server: Apache
> X-Powered-By: PHP/7.0.29
> Location: wmts.xml
> Vary: Accept-Encoding
> Content-Type: text/html; charset=UTF-8
> X-Varnish: 928654355
> Age: 0
> Via: 1.1 varnish (Varnish/6.0)
> Transfer-Encoding: chunked
> Connection: keep-alive

As this part is handled by QT itself, there is not much I can do about it. I won't definitely not be patching QT or re-implementing some of its parts with workarounds because of some broken WMTS server. Please report the issue to the server owner.

**Comment by tumic0** [comment]
I have looked deeper into the code and I can do a workaround for such broken servers. But I would rather see it fixed on the server side...

**Comment by sikmir** [comment]
I'll try to contact Mats Troeng if possible.

**Comment by sikmir** [comment]
RFC 2616 has been replaced by RFC 7231, since June 2014.
[7.1.2.  Location](https://tools.ietf.org/html/rfc7231#section-7.1.2):

>   Location = URI-reference
> 
>    The field value consists of a single URI-reference.  When it has the
>    form of a relative reference ([RFC3986], Section 4.2), the final
>    value is computed by resolving it against the effective request URI
>    ([RFC3986], Section 5).

**Comment by sikmir** [comment]
As a workaround the problem fixed on server side (the redirect permanently removed), thanks to Jarkko and Mats.

**Comment by tumic0** [comment]
They have indeed changed the standard in this case (which is IMHO a really bad idea as you will end with incompatible old clients)... Nevertheless I would implement this then, but after the "workaround" I don't have the server to test it against...

**Comment by tumic0** [comment]
They still use the redirect for the "to be explored" tiles, which causes GPXSee to do "wild things" with the requests. I have fixed this in 5.9, that will be released ASAP, but a few GPXSee < 5.9 users can now IMHO create a "deadly traffic" to the map (was not possible before as they were "blocked" by the capabilities redirect)...
